de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/generator/util/ReferenceTargetHelperTest.xtend

```diff
@@ -0,0 +1,199 @@
+	}
+	
+	@Test
+	def void testImportedModelIsProxy() {
+		val testImportTarget = testTargetResource.contents.get(0) as TargetModel
+		importedModelResource.contents.get(0) as TargetModel
+		ReferenceTargetHelper.importedModelIsProxy(testImportTarget)
+		
+		Assert.assertFalse("Imported models can be resolved.", false)
+		
+		Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().put("tmodel_inv", resourceFactory);
+		val proxyTargetResource = rs.getResource(uriProxyTarget,true)
+		val proxyTarget = proxyTargetResource.contents.get(0) as TargetModel
+		val modelIsProxy = ReferenceTargetHelper.importedModelIsProxy(proxyTarget)
+		
+		Assert.assertTrue("Imported models are proxy.", modelIsProxy)
+	}
+	
+	@Test
+	def void testParentIsProxy() {
+		val testTarget = testTargetResource.contents.get(0) as TargetModel
+		parentTargetResource.contents.get(0) as TargetModel
+		ReferenceTargetHelper.parentIsProxy(testTarget)
+
+		Assert.assertFalse("ParentTarget can be resolved", false)
+		
+		Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().put("tmodel_inv", resourceFactory);
+		val proxyTargetResource = rs.getResource(uriProxyTarget,true)
+		val proxyTarget = proxyTargetResource.contents.get(0) as TargetModel
+		val parentIsProxy = ReferenceTargetHelper.parentIsProxy(proxyTarget)
+
+		Assert.assertTrue("ParentTarget is proxy", parentIsProxy)
+	}
+	
+	@Test
+	def void testFindTargetfileOfTmodel() {
+		
+		val outputDirectory = "./target"
+
+		val workspace = ResourcesPlugin.getWorkspace();
+		val root = workspace.getRoot();
+		val project  = root.getProject("testProject");
+		val folder = project.getFolder("target");
+		val tmodelFile = folder.getFile("target.tmodel");
+		project.create(null);
+		if (!project.isOpen()) { 
+			project.open(null)
+		}
+		folder.create(IResource.NONE, true, null);
+		val bytes = "
+			Target target {
+				
+			}".getBytes();
+		val source = new ByteArrayInputStream(bytes);
+		tmodelFile.create(source, IResource.NONE, null);
+		
+		Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().put("tmodel", resourceFactory);
+		val testTargetResource = rs.getResource(URI.createPlatformResourceURI("/testProject/target/target.tmodel", true), true)
+		val target = testTargetResource.contents.get(0) as TargetModel
+		
+		val targetFile = folder.getFile("target.target");
+		val bytesTarget = 
+			'''<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+			<?pde version="3.8"?><target name="target" sequenceNumber="1">
+			<locations>
+			</locations>
+			<environment>
+			<os></os>
+			<ws>gtk</ws>
+			<arch></arch>
+			<nl></nl> 
+			</environment>
+			<targetJRE path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/"/>
+			</target>'''.toString.getBytes();
+		val sourceTarget = new ByteArrayInputStream(bytesTarget);
+		targetFile.create(sourceTarget, false, null);
+		
+		Assert.assertNotNull(ReferenceTargetHelper.findTargetfileOfTmodel(target, outputDirectory))

```

Comment:
Maybe also assert that the file you find is the one you created before... "NotNull" can be quite a lot... ;)